### PR TITLE
Prune user auths after instruction execution (settlement)

### PR DESCRIPTION
**Pull request by maxsam4**

### CODE REVIEW

File: pallets/settlement/src/lib.rs [hunk 1]
```diff
@@ -1164,15 +1164,30 @@ impl<T: Trait> Module<T> {
...
         (
             instructions_processed,
             weight_for_execution.saturating_add(T::DbWeight::get().writes(5)),
         )
     }
 
+    fn prune_user_auths(legs: &Vec<(u64, Leg<T::Balance>)>, instruction_id: u64) {
+        // We remove duplicates in memory before triggering storage actions
+        let mut counter_parties = Vec::with_capacity(legs.len() * 2);
+        for (_id, leg) in legs {
```
Comment (Centril) [inline pallets/settlement/src/lib.rs:1180]:
```suggestion
        for (_, leg) in legs {
```

Comment (maxsam4) [reply pallets/settlement/src/lib.rs:1180]: I prefer to name variables even if they are unused. They help in providing context.

Comment (Centril) [reply pallets/settlement/src/lib.rs:1180]: Context can be inferred from `legs`'s type. `_id` doesn't add anything functional here, so imo its just noise.

Comment (maxsam4) [reply pallets/settlement/src/lib.rs:1180]: I don't agree but I don't really care about this either. Will commit the change.

**Review by Centril — APPROVED** [review]

**Review by satyamakgec — APPROVED** [review]
I don't see a change for Authorized to Unknown in settlement runtime.

**Comment by maxsam4** [comment]
> I don't see a change for Authorized to Unknown in settlement runtime.

We are now pruning this data after the settlement. Therefore, the data (Authorized) reverts to the default value (Unknown).

**Review by adamdossa — APPROVED** [review]

**Pull request merged**